engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/multiinstance/MultiInstanceActivityTest.java

```diff
@@ -1291,6 +1294,74 @@ public void shouldTriggerNonInterruptingBoundaryEvent() {
             tuple(PROCESS_ID, BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));
   }
 
+  @Test
+  public void shouldSetNumberOfElementInstancesVariable() {
+    // given
+    ENGINE.deployment().withXmlResource(process(miBuilder)).deploy();
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariable(INPUT_COLLECTION_EXPRESSION, INPUT_COLLECTION)
+            .create();
+
+    // when
+    completeJobs(processInstanceKey, INPUT_COLLECTION.size());
+
+    // then
+    final long flowScopeKey =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementType(BpmnElementType.MULTI_INSTANCE_BODY)
+            .getFirst()
+            .getKey();
+
+    final var variableRecord =
+        RecordingExporter.variableRecords()
+            .withName(NUMBER_OF_ELEMENT_INSTANCES)
+            .withScopeKey(flowScopeKey)
+            .withIntent(VariableIntent.CREATED)
+            .getFirst();
+
+    assertThat(variableRecord.getValue()).hasValue(String.valueOf(INPUT_COLLECTION.size()));
+  }
+
+  @Test
+  public void shouldSetNumberOfActiveElementInstancesVariable() {
+    // given
+    ENGINE.deployment().withXmlResource(process(miBuilder)).deploy();
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariable(INPUT_COLLECTION_EXPRESSION, INPUT_COLLECTION)
+            .create();
+
+    // when
+    final int completedJobs = 2;
+    completeJobs(processInstanceKey, completedJobs);
+
+    // then
+    final long flowScopeKey =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementType(BpmnElementType.MULTI_INSTANCE_BODY)
+            .getFirst()
+            .getKey();
+
+    final var variableRecord =
+        RecordingExporter.variableRecords()
+            .withName(NUMBER_OF_ACTIVE_ELEMENT_INSTANCES)
+            .withScopeKey(flowScopeKey)
+            .withIntent(VariableIntent.UPDATED)
+            .getLast();
+
+    assertThat(variableRecord.getValue())
+        .hasValue(String.valueOf(INPUT_COLLECTION.size() - completedJobs));
+  }
```

Comment:
👍 This test case is already good on it's own

❌ We should however consider how to test that the value is kept up to date with the actual number of active element instances. For example by adding an additional test, like this one, that checks the value after each completed job. 

🔧  When you've added a test case to verify that it should be updated, then you could consider changing the test you already wrote to verify its initial value (with Variableintent.CREATED instead of UPDATED).

🔧  Personally I appreciate group these tests near the `shouldSetLoopCounterVariable` testcase.